runtime/manta/tests/xcm_mock/mod.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020-2023 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(test)]
+
+pub mod parachain;
+pub mod relay_chain;
+pub mod xcm_tests;
+
+use frame_support::traits::GenesisBuild;
+use polkadot_parachain::primitives::Id as ParaId;
+use sp_runtime::traits::AccountIdConversion;
+use xcm_simulator::{decl_test_network, decl_test_parachain, decl_test_relay_chain};
+pub const ALICE: sp_runtime::AccountId32 = sp_runtime::AccountId32::new([0u8; 32]);
+pub const INITIAL_BALANCE: u128 = 10_000_000_000_000_000;
+pub const PARA_A_ID: u32 = 1;
+pub const PARA_B_ID: u32 = 2;
+pub const PARA_C_ID: u32 = 3;
+
+decl_test_parachain! {
+    pub struct ParaA {
+        Runtime = parachain::Runtime,
+        XcmpMessageHandler = parachain::XcmpQueue,
+        DmpMessageHandler = parachain::MsgQueue,
+        new_ext = para_ext(PARA_A_ID),
+    }
+}
+
+decl_test_parachain! {
+    pub struct ParaB {
+        Runtime = parachain::Runtime,
+        XcmpMessageHandler = parachain::XcmpQueue,
+        DmpMessageHandler = parachain::MsgQueue,
+        new_ext = para_ext(PARA_B_ID),
+    }
+}
+
+decl_test_parachain! {
+    pub struct ParaC {
+        Runtime = parachain::Runtime,
+        XcmpMessageHandler = parachain::XcmpQueue,
+        DmpMessageHandler = parachain::MsgQueue,
+        new_ext = para_ext(PARA_C_ID),
+    }
+}
+
+decl_test_relay_chain! {
+    pub struct Relay {
+        Runtime = relay_chain::Runtime,
+        XcmConfig = relay_chain::XcmExecutorConfig,
+        new_ext = relay_ext(),
+    }
+}
```

Comment:
Could use real polkadot runtime, applies to Calamari runtime too. Like here: https://github.com/AcalaNetwork/Acala/blob/54feaa65d729ea1e4f269d52fd4cbb2d0e304a48/runtime/integration-tests/src/relaychain/kusama_test_net.rs#L31

Could be a followup issue to this PR as well